### PR TITLE
CreateCase Callback Functional Test

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-744

### Change description ###
Added CreateCaseTest func test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+     */
+    private String invokeCallbackEndpoint(
+        CaseDetails exceptionRecord
+    ) throws IOException {
+        CaseDetails exceptionRecordWithSearchFields =
+            exceptionRecord.toBuilder().build();
+
+        CallbackRequest callbackRequest = CallbackRequest
+            .builder()
+            .eventId("createNewCase")
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: there might be a constant for that

File: charts/bulk-scan-orchestrator/values.preview.template.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ java:
     DELETE_ENVELOPES_DLQ_MESSAGES_ENABLED: "true"
     DELETE_ENVELOPES_DLQ_MESSAGES_CRON: "0 * * * * *"
     DELETE_ENVELOPES_DLQ_MESSAGES_TTL: "10s"
+    TRANSFORMATION_URL_BULKSCAN: "http://bulk-scan-sample-app-aat.service.core-compute-aat.internal"
```
Comment (doncem) [inline charts/bulk-scan-orchestrator/values.preview.template.yaml]: unfortunately, this will have to be a separate pr as it is a config change. only minor one.

but if we're happy to stick this in together @nitinprabhu then let's bump chart version here too and we are all good

Comment (nitinprabhu) [reply charts/bulk-scan-orchestrator/values.preview.template.yaml]: Yes as commented on pr earlier this should go as a separate pr and bump chart version including flux repo

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 1 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Done

File: charts/bulk-scan-orchestrator/values.preview.template.yaml (hunk 2 above)
Comment (apereverzin) [reply charts/bulk-scan-orchestrator/values.preview.template.yaml]: Everything done - PR-660

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                "case.poBox", poBox.toString()
+            )
+        );
+        return caseDetailsList.stream().findFirst();
+    }
+
+    private Boolean isCaseCreated(String caseCcdId, CaseDetails exceptionRecord) {
+
+        CaseDetails createdCase = ccdApi.getCase(
+            String.valueOf(caseCcdId),
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: it's a string already

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            exceptionRecord.getJurisdiction()
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: you could pass just the jurisdiction to this method

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        CaseDetails exceptionRecord = createExceptionRecord("envelopes/new-envelope-create-case-with-evidence.json");
+
+        // when
+        String caseCcdId = invokeCallbackEndpoint(exceptionRecord);
+
+        // then
+        await("Case is created")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> isCaseCreated(caseCcdId, exceptionRecord));
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: There's no need to wait for the case to be created - it is created at this point. We only wait when cases are supposed to be created as a result of processing a queue message.

It would be great, though, to verify the details of the exception record.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            SampleData.JURSIDICTION,
+            SampleData.JURSIDICTION + "_" + CreateExceptionRecord.CASE_TYPE,
+            ImmutableMap.of(
+                "case.poBox", poBox.toString()
+            )
+        );
+        return caseDetailsList.stream().findFirst();
+    }
+
+    private Boolean isCaseCreated(String caseCcdId, CaseDetails exceptionRecord) {
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Is we remove the waiting for the case to be created, this method can also go.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        // given
+        CaseDetails exceptionRecord = createExceptionRecord("envelopes/new-envelope-create-case-with-evidence.json");
+
+        // when
+        String caseCcdId = invokeCallbackEndpoint(exceptionRecord);
+
+        // then
+        CaseDetails createdCase = ccdApi.getCase(caseCcdId, exceptionRecord.getJurisdiction());
+        assertThat(createdCase.getCaseTypeId()).isEqualTo(exceptionRecord.getCaseTypeId());
+        assertThat(createdCase.getData()).containsExactlyEntriesOf(exceptionRecord.getData());
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: I guess these two lines will fail, because the case will have a different case type Id and the data will also be different.

Case type id will be `Bulk_Scanned` and fields in the case will be same as here: https://github.com/hmcts/bulk-scan-ccd-definitions/blob/master/definitions/bulkscan/data/sheets/CaseField.json 

First name, last name and email should be the same as in the exception record's OCR data in the file.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 3 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Have fixed

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 4 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Have fixed

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java (hunk 5 above)
Comment (apereverzin) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java]: Done

File: src/functionalTest/resources/envelopes/new-envelope-create-case-with-evidence.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    {
+      "metadata_field_name": "email",
+      "metadata_field_value": "hello@test.com"
+    }
+  ],
+  "ocr_data_validation_warnings": [
+    "warning 1",
+    "warning 2"
+  ],
+  "payments": [
```
Comment (nitinprabhu) [inline src/functionalTest/resources/envelopes/new-envelope-create-case-with-evidence.json]: Should we exclude payments in the request?

Comment (lgonczar) [reply src/functionalTest/resources/envelopes/new-envelope-create-case-with-evidence.json]: This is the content of a queue message, so I guess it doesn't matter if we include them or not. Might just be clearer if they're excluded.

Comment (apereverzin) [reply src/functionalTest/resources/envelopes/new-envelope-create-case-with-evidence.json]: Done
